### PR TITLE
updated to latest api

### PR DESCRIPTION
**Pull request by Xelwon**

### CODE REVIEW

File: POGOLib.Official/Net/RpcClient.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ internal RpcClient(Session session)
         {
             _session = session;
             _rpcEncryption = new RpcEncryption(session);
-            _mapKey = string.Empty;
+            _mapKey = "90f6a704505bccac73cec99b07794993e6fd5a12"; //unknownPtr8 for .57.4
```
Comment (AeonLucid) [inline POGOLib.Official/Net/RpcClient.cs:70]: We don't have to hardcode the unknown ptr 8 value. If you keep it empty, POGO will respond with a key for you to use. See https://github.com/AeonLucid/POGOLib/blob/e24743018613ef1d0e4288e4d29baf056c02d6d0/POGOLib.Official/Net/RpcClient.cs#L608-L613.

File: POGOLib.Official/Net/Session.cs [hunk 2]
```diff
@@ -149,7 +144,9 @@ public void Shutdown()
             var result = Configuration.Hasher.PokemonVersion.CompareTo(pogoVersion);
             if (result != 0)
             {
-                throw new HashVersionMismatchException($"The version of the {nameof(Configuration.Hasher)} ({Configuration.Hasher.PokemonVersion}) does not match the minimal API version of PokemonGo ({pogoVersion}). Set 'Configuration.IgnoreHashVersion' to true if you want to disable the version check.");
+                var str1 = "Set 'Configuration.IgnoreHashVersion' to true if you want to disable the version check.";
+                var str2 = $"The version of the {nameof(Configuration.Hasher)} ({Configuration.Hasher.PokemonVersion}) does not match the minimal API version of PokemonGo ({pogoVersion}). {str1}";
+                throw new HashVersionMismatchException(str2);
```
Comment (AeonLucid) [inline POGOLib.Official/Net/Session.cs:149]: Why did you change this?

File: POGOLib.Official/Util/Hash/PokeHashHasher.cs [hunk 3]
```diff
@@ -108,15 +109,24 @@ public PokeHashHasher(string[] authKeys)
...
+                        hashData.LocationAuthHash = responseData.LocationAuthHash;
+                        hashData.LocationHash = responseData.LocationHash;
+                        var arr = new  List<ulong>();
+                        foreach (var element in responseData.RequestHashes) {
+                            arr.Add( unchecked((ulong) element));
+                        }
+                        hashData.RequestHashes = arr.ToArray();
+                        return hashData;
+                        /*
+                        return
```
Comment (AeonLucid) [inline POGOLib.Official/Util/Hash/PokeHashHasher.cs:122]: Why?

File: POGOLib.Official/Pokemon/Player.cs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿using System;
```
Comment (AeonLucid) [inline POGOLib.Official/Pokemon/Player.cs:1]: Why are you removing the documentation blocks?

File: POGOLib.Official/Net/Session.cs [hunk 5]
```diff
@@ -115,16 +114,12 @@ internal Session(ILoginProvider loginProvider, AccessToken accessToken, GeoCoord
...
+                CheckHasherVersion().Wait();
 
-            if (!await RpcClient.StartupAsync().ConfigureAwait(false))
-            {
+            if (!RpcClient.StartupAsync().Result)
                 return false;
-            }
 
-            await _heartbeat.StartDispatcher().ConfigureAwait(false);
+            _heartbeat.StartDispatcher().Wait();
```
Comment (AeonLucid) [inline POGOLib.Official/Net/Session.cs:122]: Why are you removing await? This function is async so await should be used.

**Review by AeonLucid — CHANGES_REQUESTED** [review]

File: POGOLib.Official/Net/RpcClient.cs (hunk 1 above)
Comment (Xelwon) [reply POGOLib.Official/Net/RpcClient.cs:70]: I didn`t see it.

File: POGOLib.Official/Net/Session.cs (hunk 5 above)
Comment (Xelwon) [reply POGOLib.Official/Net/Session.cs:122]: because await is to do it syncronous, configureawait(false) to use same threadspace.
so you can replace it using the method .wait(), (or .Resut if return something) leaving the code cleaner.

File: POGOLib.Official/Net/Session.cs (hunk 2 above)
Comment (Xelwon) [reply POGOLib.Official/Net/Session.cs:149]: that was not importan change, some, text editors seeing that the lenght of that string is greater than allow by them.

File: POGOLib.Official/Util/Hash/PokeHashHasher.cs (hunk 3 above)
Comment (Xelwon) [reply POGOLib.Official/Util/Hash/PokeHashHasher.cs:122]: change unneeded, yes, I did it to can trace it better.

File: POGOLib.Official/Pokemon/Player.cs (hunk 4 above)
Comment (Xelwon) [reply POGOLib.Official/Pokemon/Player.cs:1]: change unneeded too,  I deleted it only to do it cleaner 👍
